src/prisma/_transactions.py

```diff
@@ -3,23 +3,39 @@
 import logging
 import warnings
 from types import TracebackType
-from typing import TYPE_CHECKING, Generic, TypeVar
+from typing import TYPE_CHECKING, Any, Generic, TypeVar
 from datetime import timedelta
 
 from ._types import TransactionId
 from .errors import TransactionNotStartedError
+from ._compat import StrEnum
 from ._builder import dumps
 
 if TYPE_CHECKING:
     from ._base_client import SyncBasePrisma, AsyncBasePrisma
 
 log: logging.Logger = logging.getLogger(__name__)
 
+__all__ = (
+    'TransactionIsolationLevel',
+    'AsyncTransactionManager',
+    'SyncTransactionManager',
+)
+
 
 _SyncPrismaT = TypeVar('_SyncPrismaT', bound='SyncBasePrisma')
 _AsyncPrismaT = TypeVar('_AsyncPrismaT', bound='AsyncBasePrisma')
 
 
+# See here: https://www.prisma.io/docs/orm/prisma-client/queries/transactions#supported-isolation-levels
+class TransactionIsolationLevel(StrEnum):
```

Comment:
I'm somewhat torn between:
- Just defining this enum like you have with all options present
- Auto generating the enum to only include options that are supported by the current database

In general I've been leaning towards limiting code generation as much as possible but it might be worth it here? Do you have any opinions?

Reply:
If a user tries to start a transaction with an invalid/unsupported isolation level, they will receive a `500 Internal Server Error` from the Prisma, but the engine handle this case and raises [InvalidIsolationLevel](https://github.com/prisma/prisma-engines/blob/5.8.0/quaint/src/error.rs#L268-L269) error. I'm not sure, but it seems that `prisma-ts-client` should generate the enum.

Reply:
@RobertCraigie Finally, I chose this option:

> Auto generating the enum to only include options that are supported by the current database

PR updated.